### ISSUE TITLE
请问一下这个llm第一轮预训练loss没怎么下降是正常的吗，训练代码和模型代码没有用作者的，数据用的也是通用猴子序列，想参考一下这个曲线健不健康，谢谢

### ISSUE THREAD

**Issue by wz043** [qibin0506/Cortex#21]
<img width="1529" height="518" alt="Image" src="https://github.com/user-attachments/assets/6986c826-647d-458c-81fb-2fe709ffb3ab" />
模型大概0.7B，然后最大学习率是1e-4，余弦衰减，优化器是AdamW，batch设置的是9，梯度裁剪是1.0，

[train.py](https://github.com/user-attachments/files/22309235/train.py)

**Comment by qibin0506** [comment]
看着不太正常，loss基本没什么变化，模型没有学到东西，应该是完全随机的状态

**Comment by wz043** [comment]
> 看着不太正常，loss基本没什么变化，模型没有学到东西，应该是完全随机的状态

是不是学习率我设置的太低了，还是说模型有问题呢，请问一下你们当时预训练第一个eporch经过后loss大概是下降到多少呢

[inference_model.py](https://github.com/user-attachments/files/22310033/inference_model.py)

**Comment by wz043** [comment]
> 看着不太正常，loss基本没什么变化，模型没有学到东西，应该是完全随机的状态

那再请问一下这个模型架构最后这里如果权重指定一致会有问题吗，self.head.weight = self.emb.emb.weight
        self.register_buffer("freqs_cis", precompute_freqs_cis(args), persistent=False)

    def forward(self, input_ids, attention_mask, start_pos: int = 0, **kwargs):
        pad_mask = attention_mask
        tokens = input_ids
        seqlen = tokens.size(1)
        freqs_cis = self.freqs_cis[start_pos:start_pos + seqlen]
        h = self.emb(tokens)
        mask = None
        if seqlen > 1:
            mask = torch.full((seqlen, seqlen), float("-inf"), device=tokens.device).triu_(1)
        for layer in self.layers:
            h = layer(h, start_pos, freqs_cis, mask, pad_mask)
        h = self.norm(h)
        logits = self.head(h)
        return logits，其他人建议把这里用和词嵌一样的权重
